### PR TITLE
Add Travis CI build.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+language: java
+
+jdk:
+  - oraclejdk7
+  - oraclejdk8
+
+install: mvn -f gson install -DskipTests=true
+script: mvn -f gson test
```
Comment (JakeWharton) [inline .travis.yml]: Would love to see the repo updated to a more traditional Java repo layout and some of the other cruft removed or consolidated.

Comment (inder123) [inline .travis.yml]: Do you mean move the sub-projects out as their own projects? What would you suggest?

Comment (JakeWharton) [inline .travis.yml]: I looked at it more after writing this, and didn't realize that each folder was a separate Maven-based project. With that, a simple parent pom that includes each folder as a module will probably be enough to allow everything to be built at once.

Comment (inder123) [inline .travis.yml]: True. However, other sub-projects are in states of disrepair and not maintained well. We didn't want to tie them in with Gson.

**Pull request merged**